packages/store-sync/src/recs/defineInternalComponents.ts

```diff
@@ -0,0 +1,30 @@
+import { World, defineComponent, Type, Component } from "@latticexyz/recs";
+import { Table } from "../common";
+
+function typedComponent<TType>(
+  component: Component
+): typeof component extends Component<infer TSchema, infer TMetadata> ? Component<TSchema, TMetadata, TType> : never {
+  return component;
+}
```

Comment:
I am trying to change the position of the custom type to allow `Type.T` to resolve as a strong type, while inferring the schema/metadata without having to define them twice in `defineComponent` (once in the type, once in the object).

Without wrapping, I get the inferred schema/metadata as expected:
<img width="365" alt="image" src="https://github.com/latticexyz/mud/assets/508855/579e7992-59c7-41f0-9498-fdec092f8200">

But once I wrap, it looks like:
<img width="568" alt="image" src="https://github.com/latticexyz/mud/assets/508855/2dce7dd8-826a-4c8e-810b-4153fafd49ef">

I also tried a separate `defineTypedComponent` that takes in different order of type parameters, using the same inference as `defineComponent`, but that did the same thing.
<img width="830" alt="image" src="https://github.com/latticexyz/mud/assets/508855/3cb9b04a-f375-4b86-b7e5-8df080a26afa">